sentry_sdk/integrations/django/caching.py

```diff
@@ -1,80 +1,152 @@
 import functools
 from typing import TYPE_CHECKING
+from urllib3.util import parse_url as urlparse
 
 from django import VERSION as DJANGO_VERSION
 from django.core.cache import CacheHandler
 
 import sentry_sdk
 from sentry_sdk.consts import OP, SPANDATA
-from sentry_sdk.utils import ensure_integration_enabled
+from sentry_sdk.utils import (
+    SENSITIVE_DATA_SUBSTITUTE,
+    capture_internal_exceptions,
+    ensure_integration_enabled,
+)
 
 
 if TYPE_CHECKING:
     from typing import Any
     from typing import Callable
+    from typing import Optional
 
 
 METHODS_TO_INSTRUMENT = [
+    "set",
+    "set_many",
     "get",
     "get_many",
 ]
 
 
-def _get_span_description(method_name, args, kwargs):
-    # type: (str, Any, Any) -> str
-    description = "{} ".format(method_name)
-
+def _get_key(args, kwargs):
+    # type: (list[Any], dict[str, Any]) -> str
     if args is not None and len(args) >= 1:
-        description += str(args[0])
+        key = args[0]
+        if isinstance(key, dict):
+            # Do not leak sensitive data
+            # `set_many()` has a dict {"key1": "value1", "key2": "value2"} as first argument.
+            # Those values could include sensitive data so we replace them with a placeholder
+            key = {x: SENSITIVE_DATA_SUBSTITUTE for x in key}
+        return str(key)
     elif kwargs is not None and "key" in kwargs:
-        description += str(kwargs["key"])
+        return str(kwargs["key"])
```

Comment:
Why do we do the sensitive data stripping above but not here?

Reply:
I refactored the function to always strip sensitive data if it is a dict